### PR TITLE
[avcpp] Update to 2.4.0

### LINKED ISSUE #38582: [avcpp] Can't use avcpp with CMake on x64-windows because CMake can't find Threads::Threads
**Describe the bug**
As the title says. The build itself is successful so the issue must be in the provided CMake files. On Linux it works

**Environment**
- OS: Windows 11
- Compiler: MSVC 19.37.32822 under Visual Studio Community 2022 17.7.3

**To Reproduce**
Add "avcpp" to vcpkg.json dependencies and in CMakeLists.txt `find_package(avcpp CONFIG REQUIRED)`, then link `avcpp::avcpp` and `avcpp::FFmpeg` to your target. I used it in Visual Studio (opening the project with "open folder") but I expect the same to happen with standalone CMake. I'm using vcpkg in manifest mode as a git submodule

**Expected behavior**
It should work

**Failure logs**
CMake failure:
```
1> [CMake] CMake Error at build/debug/vcpkg_installed/x64-windows/share/avcpp/avcpp-targets.cmake:61 (set_target_properties):
1> [CMake]   The link interface of target "avcpp::avcpp" contains:
1> [CMake] 
1> [CMake]     Threads::Threads
1> [CMake] 
1> [CMake]   but the target was not found.  Possible reasons include:
1> [CMake] 
1> [CMake]     * There is a typo in the target name.
1> [CMake]     * A find_package call is missing for an IMPORTED target.
1> [CMake]     * An ALIAS target is missing.
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   build/debug/vcpkg_installed/x64-windows/share/avcpp/avcpp-config.cmake:26 (include)
1> [CMake]   vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
1> [CMake]   CMakeLists.txt:16 (find_package)
```

### PR DESCRIPTION
**Pull request by MonicaLiu0311**
Fixes #38582. PR https://github.com/h4tr3d/avcpp/pull/137 submitted to upstream has been merged, and a new version has been released by upstream, so updating to 2.4.0 fixes the issue:
```
1> [CMake] CMake Error at build/debug/vcpkg_installed/x64-windows/share/avcpp/avcpp-targets.cmake:61 (set_target_properties):
1> [CMake]   The link interface of target "avcpp::avcpp" contains:
1> [CMake] 
1> [CMake]     Threads::Threads
1> [CMake] 
1> [CMake]   but the target was not found.  Possible reasons include:
1> [CMake] 
1> [CMake]     * There is a typo in the target name.
1> [CMake]     * A find_package call is missing for an IMPORTED target.
1> [CMake]     * An ALIAS target is missing.
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   build/debug/vcpkg_installed/x64-windows/share/avcpp/avcpp-config.cmake:26 (include)
1> [CMake]   vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
1> [CMake]   CMakeLists.txt:16 (find_package)
```

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] ~SHA512s are updated for each updated download.~
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/avcpp/portfile.cmake [hunk 1]
```diff
@@ -32,6 +32,12 @@ vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/${PORT}")
 
 vcpkg_fixup_pkgconfig()
 
+file(READ "${CURRENT_PACKAGES_DIR}/share/${PORT}/${PORT}-config.cmake" _contents)
+file(WRITE "${CURRENT_PACKAGES_DIR}/share/${PORT}/${PORT}-config.cmake" "
+include(CMakeFindDependencyMacro)
+find_dependency(Threads)
+${_contents}")
```
Comment (Neumann-A) [inline ports/avcpp/portfile.cmake:39]: Did you check the file after this?

Comment (MonicaLiu0311) [reply ports/avcpp/portfile.cmake]: Yes, usage was tested as well:

image

Comment (dg0yt) [reply ports/avcpp/portfile.cmake]: For a permanent fix, this should be formed into a proper patch and upstreamed.

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by dg0yt** [comment]
Again: The proper fix can and shall be upstreamed. Which means there will be a patch file available.

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by data-queue** [comment]
Can you open a PR upstream?

**Comment by MonicaLiu0311** [comment]
> Can you open a PR upstream?

Done🌻

**Pull request merged**